src/core/Plugin_social.ml

```diff
@@ -145,29 +145,62 @@ let print_diff (f:float) : string =
     [spf "%d seconds" s];
   ] |> List.flatten |> String.concat ", "
 
+
+let create_message_for_user now (user,last) =
+  let diff = now -. last in
+  CCFormat.sprintf "seen %s last: %s ago" user (print_diff diff)
+
 let cmd_seen (state:state) =
-  Command.make_simple
+  Command.make_simple_l
     ~descr:"ask for the last time someone talked on this chan"
     ~prio:10 ~prefix:"seen"
     (fun _msg s ->
        try
-         let dest = String.trim s in
+         let dest = String.trim s |> String.uppercase in
          Log.logf "query: seen `%s`" dest;
-         begin match StrMap.get dest state.map with
-           | Some data ->
-             let last = data.last_seen in
-             let now = Unix.time () in
-             let diff = now -. last in
-             let msg =
-               CCFormat.sprintf "seen %s last: %s ago" dest (print_diff diff)
-             in
-             Lwt.return_some msg
-           | None ->
-             Lwt.return_some "who?"
-         end
+         let now = Unix.time () in
+         StrMap.fold (fun name data acc ->
+             if dest = String.uppercase name then
+               (name, data.last_seen) :: acc
+             else
+               acc
+           ) state.map []
+         |> CCList.sort (fun a b -> - (compare (snd a) (snd b)) )
+         |> CCList.map (create_message_for_user now)
+         |> Lwt.return
        with e ->
          Lwt.fail (Command.Fail ("seen: " ^ Printexc.to_string e)))
 
+
+let cmd_lastseen (state:state) =
+  Command.make_simple_l
+    ~descr:"ask for the last time someone talked on this chan"
```

Comment:
shouldn't it be "last_talked", then? "seen" would be confusing otherwise

Reply:
What about just making it !last then?